lib/util/hasNonEmptyProp.js

```diff
@@ -12,7 +12,16 @@ var getProp = require("jsx-ast-utils").getProp;
  * @returns boolean
  */
 function hasNonEmptyProp(attributes, name) {
-    return hasProp(attributes, name) && getPropValue(getProp(attributes, name)).trim().length > 0;
+    const propValue = getPropValue(getProp(attributes, name));
+    /**
+     * getPropValue normalized "true" and "false" to true and false boolean.
+     * It is sufficent to check is the prop exists in such case.
+     */
```

Comment:
library code reference for this: https://github.com/jsx-eslint/jsx-ast-utils/blob/main/src/values/Literal.js